### PR TITLE
Fuzzer performs multiple operations on the underlying array instead of just one

### PR DESCRIPTION
**Pull request by robert3005**
Includes fixes for
- ZigZagArray scalar_at returns wrong nullability
- RunEndArray slice wrongly slicing validity
- Scalar comparison is nullability invariant

### CODE REVIEW

File: fuzz/src/lib.rs [hunk 1]
```diff
@@ -29,55 +58,378 @@ pub enum Action {
...
-            Some(u.int_in_range(min..=max).map(|i| i as u64))
+            Some(u.int_in_range(min..=max))
         } else {
             None
         }
     })
     .collect::<Result<Vec<_>>>()
 }
+
+fn slice_primitive_array(array: &Array, start: usize, stop: usize) -> Array {
```
Comment (robert3005) [inline fuzz/src/lib.rs]: Should split these functions from this file

File: vortex-array/src/canonical.rs [hunk 2]
```diff
@@ -89,6 +89,18 @@ impl Canonical {
             }
         }
     }
+
+    /// Unwrap Canonical Array back into a regular Array
+    pub fn into_array(self) -> Array {
```
Comment (lwwmanning) [inline vortex-array/src/canonical.rs]: why this vs just implementing the `IntoArray` trait for Canonical?

Comment (robert3005) [reply vortex-array/src/canonical.rs]: should do just that

Comment (robert3005) [reply vortex-array/src/canonical.rs]: This already exists

**Review by lwwmanning — APPROVED** [review]

**Review by AdamGS — APPROVED** [review]

**Pull request merged**